packages/react-widgets/src/widgets/FeatureSelectionWidget.js

```diff
@@ -0,0 +1,171 @@
+  <SvgIcon>
+    <path d='M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 1 0 0 16 8 8 0 0 0 0-16zm0 6a2 2 0 1 1 0 4 2 2 0 0 1 0-4z' />
+  </SvgIcon>
+);
+
+const LassoIcon = (
+  <SvgIcon>
+    <path d='M12 3c4.935 0 9 3.736 9 9l-.002.343-.012.669c-.012.437-.033.86-.062 1.265l-.05.597C20.498 18.767 19.267 21 17 21c-1.192 0-1.971-.341-2.987-1.122l-.473-.375c-.401-.319-.64-.473-.888-.566a4.894 4.894 0 0 0-.415-.13l-.34-.085-.398-.086-.456-.086-.66-.111-1.708-.273a9.103 9.103 0 0 1-.952-.206C5.46 17.301 3 14.954 3 12.015c0-1.508.485-2.995 1.436-4.458.355.585.906 1.04 1.562 1.272C5.328 9.916 5 10.977 5 12.015c0 1.889 1.78 3.588 3.282 4.025l.085.023.345.076.517.092 1.619.257.583.1.518.099.237.05.433.102c.272.07.512.143.73.224.434.161.783.373 1.235.718l.457.362c.806.646 1.24.857 1.959.857.893 0 1.63-1.518 1.895-4.45l.045-.585c.013-.2.024-.407.033-.62l.02-.655c.005-.224.007-.454.007-.69 0-4.12-3.134-7-7-7a1 1 0 0 1 0-2zM7 4a2 2 0 1 1 0 4 2 2 0 0 1 0-4z' />
+  </SvgIcon>
+);
+
+const DRAW_MODES_MAP = {
+  [DRAW_MODES_KEYS.POLYGON]: {
+    label: 'Polygon',
+    icon: PolygonIcon
+  },
+  [DRAW_MODES_KEYS.RECTANGLE]: {
+    label: 'Rectangle',
+    icon: RectangleIcon
+  },
+  [DRAW_MODES_KEYS.CIRCLE]: {
+    label: 'Circle',
+    icon: CircleIcon
+  },
+  [DRAW_MODES_KEYS.LASSO_TOOL]: {
+    label: 'Lasso tool',
+    icon: LassoIcon
+  }
+};
+
+const EDIT_MODES_MAP = {
+  [EDIT_MODES_KEYS.EDIT]: {
+    label: 'Edit geometry',
+    icon: CursorIcon
+  }
+};
+
+function FeatureSelectionWidget({
+  className,
+  drawModes: drawModesKeys,
+  editModes: editModesKeys,
+  defaultActivated,
+  defaultSelectedMode,
+  tooltipPlacement
+}) {
+  const dispatch = useDispatch();
+  const [activated, setActivated] = useState(defaultActivated);
+  const [selectedMode, setSelectedMode] = useState(defaultSelectedMode);
+  const geometry = useSelector((state) => state.carto.featureSelectionState.geometry);
+
+  const drawModes = useMemo(() => {
+    return drawModesKeys
+      .filter((key) => DRAW_MODES_MAP[key])
+      .map((key) => ({ id: key, ...DRAW_MODES_MAP[key] }));
+  }, [drawModesKeys]);
+
+  const editModes = useMemo(() => {
+    return editModesKeys
+      .filter((key) => EDIT_MODES_MAP[key])
+      .map((key) => ({ id: key, ...EDIT_MODES_MAP[key] }));
+  }, [editModesKeys]);
+
+  const handleActivatedChange = (newActivated) => {
+    dispatch(setFeatureSelectionMode(newActivated ? selectedMode : null));
+    setActivated(newActivated);
+  };
+
+  useEffect(() => {
+    // After drawing, when the geometry is created, disactivated the mode
+    const wasGeometryRecentlyCreated = geometry?.properties.disabled === undefined;
+    if (DRAW_MODES_MAP[selectedMode] && wasGeometryRecentlyCreated) {
+      handleActivatedChange(false);
+    }
+    // If selectedMode is added, each time the mode changes,
+    // it breaks the activated by default behavior
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I can see that you have disabled this rule `eslint-disable-next-line react-hooks/exhaustive-deps` in many places. It's recommend not to do that

Reply:
I've being trying to avoid disabling this rule. But in order to remove this one I would have to bring activated state to redux, and I don't want to add another state variable to redux.